File: src/main/java/net/siegemc/core/items/items/TestBow.java
package net.siegemc.core.items.items;

import net.siegemc.core.Core;
import net.siegemc.core.items.BowItem;
import net.siegemc.core.items.Rarity;
import net.siegemc.core.items.Stat;
import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;

import java.util.Arrays;

public class TestBow extends BowItem {
    public TestBow() {
        super("Debug Bow", new ItemStack(Material.BOW, 1));
        this.setDescription(Arrays.asList("The bow used to test if damage", "can work with BowItem."));
        this.setRarity(Rarity.DEBUG);
        this.setLevelRequirement(5);
        this.setDamage(95);
        this.getStats().put(Stat.STRENGTH, 100);
        Core.getItems().put("Debug Bow", this);
    }
}

File: src/main/kotlin/net/siegemc/core/items/implemented/weapons/ranged/TestBow.kt
package net.siegemc.core.items.implemented.weapons.ranged

import net.siegemc.core.items.CustomItemUtils
import net.siegemc.core.items.enums.Rarity
import net.siegemc.core.items.recipes.recipes
import net.siegemc.core.items.types.weapons.CustomBow
import org.bukkit.Material
import org.bukkit.inventory.ItemStack

class TestBow() : CustomBow(
    name = "Test Bow",
    customModelData = 1,
    description = listOf("A bow for testing"),
    levelRequirement = 0,
    material = Material.BOW,
    baseStats = CustomItemUtils.statMap(strength = 20.0),
    recipeList = recipes {

    }
) {

    constructor(quality: Int): this() {
        this.quality = quality
        this.rarity = Rarity.getFromInt(quality)
    }

    constructor(item: ItemStack): this() {
        this.item = item
        deserialize()
    }

}